### PR TITLE
Add support for backup verification

### PR DESCRIPTION
**Pull request by WadeBarnes**
Script and documentation updates.

### CODE REVIEW

**Comment by WadeBarnes** [comment]
Hey @jefkel, When you have a moment, could you give this a review.  Thanks

**Comment by WadeBarnes** [comment]
@jefkel, Here are some results using a larger database.  The verification on the backup was done using pod storage (not mounting a volume to `/var/lib/pgsql/data/userdata`);

Status message received from The OrgBook BC (test):
Successfully backed up wallet-db:5432/tob_holder.
Backup written to /backups/daily/2019-03-07/wallet-db-tob_holder_2019-03-07_18-00-46.sql.gz.
Database Size: 50 GB
Backup Size: 12G
Elapsed time: 0h:26m:18s - Status Code: 0

Status message received from The OrgBook BC (test):
Successfully verified backup; /backups/daily/2019-03-07/wallet-db-tob_holder_2019-03-07_18-00-46.sql.gz
The restored database contained 4 tables, and is 42 GB in size.
Elapsed time: 0h:21m:20s - Status Code: 0

The pod is `backup-14-skbs7` running on `ociopf-p-190.dmz`.  I have even larger databases we could test with just to see how far things can be pushed.  Do you have a way to monitor what is going on the pod and node if I were to run the verification while you were monitoring it?

**Comment by WadeBarnes** [comment]
@jefkel, ready for another look.  Most importantly I updated the deployment configuration to mount the backup pvc to the server data volume to avoid that issue we found.

File: README.md [hunk 1]
```diff
@@ -9,6 +9,39 @@ This project provides you with a starting point for integrating backups into you
...
 Following are the instructions for running the backups and a restore.
 
+## Storage
+*Before we get too far into the the details, we're going to take a moment to discuss the most important part of the whole process - **The Storage**.*  The backup container uses two volumes, one for storing the backups and the other for restore/verification testing.  The deployment template separates them intentionally.
+
+The following sections on storage discuss the recommendations and limitations of the storage classes created specifically for the BC Government's [PathFinder](https://console.pathfinder.gov.bc.ca:8443/) environment.
+
+
+### Backup Storage Volume
+The recommended storage class for the backup volume is `nfs-storage`.  This class of storage **cannot** be auto-provisioned through the use of a deployment template.  The `PersistentVolumeClaim` declared in the supplied deployment template for the *backup volume* will purposely fail to properly provision and wire an `nfs-storage` volume if published before you manually provision your `nfs-storage` claim.
```
Comment (jefkel) [inline README.md]:
```suggestion
The recommended storage class for the backup volume is `nfs-backup`.  This class of storage **cannot** be auto-provisioned through the use of a deployment template.  The `PersistentVolumeClaim` declared in the supplied deployment template for the *backup volume* will purposely fail to properly provision and wire an `nfs-backup` volume if published before you manually provision your `nfs-backup` claim.
```

File: README.md [hunk 2]
```diff
@@ -9,6 +9,39 @@ This project provides you with a starting point for integrating backups into you
...
+
+When using `nfs-storage` you will need to provision your claims **before** you publish your deployment configuration, through either the [service catalog](https://github.com/BCDevOps/provision-nfs-apb#provision-via-gui-catalog) using the [BC Gov NFS Storage](https://github.com/BCDevOps/provision-nfs-apb/blob/master/docs/usage-gui.md) wizard, or by using the [svcat cli](https://github.com/BCDevOps/provision-nfs-apb#provision-via-svcat-cli).
```
Comment (jefkel) [inline README.md]:
```suggestion
When using `nfs-backup` you will need to provision your claims **before** you publish your deployment configuration, through either the [service catalog](https://github.com/BCDevOps/provision-nfs-apb#provision-via-gui-catalog) using the [BC Gov NFS Storage](https://github.com/BCDevOps/provision-nfs-apb/blob/master/docs/usage-gui.md) wizard, or by using the [svcat cli](https://github.com/BCDevOps/provision-nfs-apb#provision-via-svcat-cli).
```

File: README.md [hunk 3]
```diff
@@ -9,6 +9,39 @@ This project provides you with a starting point for integrating backups into you
...
+
+You'll note the name of the resulting storage claim has a random component to it (example, `bk-devex-von-bc-tob-test-xjrmkhsnshay`).  This name needs to be injected into the template and override the default value of the `BACKUP_VOLUME_NAME` parameter in order for the storage to be correctly mounted to the `/backups/` directory of the container.  This is the reasoning behind provisioning the storage **before** publishing the deployment configuration.
```
Comment (jefkel) [inline README.md]:
```suggestion
You'll note the name of the resulting storage claim has a random component to it (example, `bk-devex-von-bc-tob-test-xjrmkhsnshay`).  This name needs to be injected into the default value of the `BACKUP_VOLUME_NAME` parameter of the template **before** publishing the deployment configuration in order for the storage to be correctly mounted to the `/backups/` directory of the container.
```

File: README.md [hunk 4]
```diff
@@ -9,6 +9,39 @@ This project provides you with a starting point for integrating backups into you
...
+
+`nfs-storage` is external to the OpenShift cluster and is covered by additional backup policies.  Best of all it has the added benefit of persisting even if the entire project is deleted, allowing it to be later mounted to another project in a disaster recovery type scenario.
```
Comment (jefkel) [inline README.md]:
```suggestion
`nfs-backup` storageClass is a lower tier of storage and not considered highly available.  `read: don't use this for live application storage`, but **IS** covered by the default enterprise backup policies, and can be directly referenced for restores using the PVC name when opening a restore ticket with 7700.
```

File: README.md [hunk 5]
```diff
@@ -9,6 +9,39 @@ This project provides you with a starting point for integrating backups into you
...
+
+`nfs-storage` **cannot** be used for restore/verification.  The permissions on the underlying volume do not allow the PostgreSql server to host it's configuration and data files on a directory backed by this class of storage.
```
Comment (jefkel) [inline README.md]:
```suggestion
`nfs-backup` PVCs **cannot** be used for restore-verification.  The permissions on the underlying volume do not allow the PostgreSql server to host it's configuration and data files on a directory backed by this class of storage.
```

File: README.md [hunk 6]
```diff
@@ -9,6 +9,39 @@ This project provides you with a starting point for integrating backups into you
...
+
+Ensure you review and plan your storage requirements before provisioning.
+
+More information on provisioning `nfs-storage` here; [provision-nfs-apb](https://github.com/BCDevOps/provision-nfs-apb)
```
Comment (jefkel) [inline README.md]:
```suggestion
More information on provisioning `nfs-backup` storage here; [provision-nfs-apb](https://github.com/BCDevOps/provision-nfs-apb)
```

File: README.md [hunk 7]
```diff
@@ -9,6 +9,39 @@ This project provides you with a starting point for integrating backups into you
...
+
+### Restore/Verification Storage Volume
+The recommended (and default) storage class for the restore/verification volume is `gluster-block`.  The supplied deployment template will auto-provision this volume for you with it is published.
```
Comment (jefkel) [inline README.md]:
```suggestion
The default storage class for the restore/verification volume is `gluster-block`.  The supplied deployment template will auto-provision this volume for you with it is published.
```

**Comment by jefkel** [comment]
latest looks good.. can't wait until we can use ephemeral for the default restore/validation tests (once we have some real numbers for size limits, etc)  That will make it even more portable.

**Pull request merged**